### ISSUE TITLE
Experiment with Multi-Module Multi-JVM setup

### ISSUE THREAD

**Issue by n1hility** [wildfly/jandex#122]
Jandex needs to be built and distributed on an older JVM than the bytecode it can read and needs to be tested with. Explore some form of setup where the test data classes (potentially including the tests if required) are in a separate module with a more recent JVM, the main jandex jar itself.

**Comment by n1hility** [comment]
Also consider pulling in the maven plugin as a module

**Comment by Ladicek** [comment]
#101 is basically the same thing.

**Comment by Ladicek** [comment]
I was able to do this with a little trick. There's a Maven plugin (called `mvn-jlink-wrapper`) that can download and unzip a JDK from AdoptOpenJDK. The Maven Compiler plugin can fork off a `javac` process from given location. The Maven JAR plugin needs extra care, because it tries to be smart in presence of `module-info.class`. I used all these 3 items on the `typeannotation-test` module and it becomes completely self-contained, built by JDK 16 even if Maven is executed with JDK 8.

I'll push a branch with all this, as well as moving to SmallRye coordinates and SmallRye parent POM, as this can't be brought in via PR. All subsequent changes will come in as PRs to the `smallrye` branch. That allows working on the 3.0 release while still keeping `master` on 2.4. Transferring the project to the SmallRye org should probably happen later on, when 3.0 is ready for release.
